src/widgetastic_patternfly4/card.py

```diff
@@ -0,0 +1,59 @@
+from widgetastic.log import call_sig
+from widgetastic.utils import ParametrizedLocator
+from widgetastic.widget import ClickableMixin
+from widgetastic.widget import ParametrizedView, ParametrizedViewRequest, GenericLocatorWidget
+from widgetastic.widget import Widget, View
+from widgetastic.xpath import quote
+from itertools import count
+
+class Card(GenericLocatorWidget):
+
+    def __init__(self, parent, locator=None, logger=None, **kwargs):
+        View.__init__(self, parent, logger=logger, **kwargs)
+        self.locator = locator or ".//article[contains(@class, 'pf-c-card')]"
+    ROOT = ParametrizedLocator("{@locator}")
+
+
+class CardForCardGroup(ParametrizedView ,Card):
+
+    def __init__(self, parent, locator=None, logger=None, **kwargs):
+        View.__init__(self, parent, logger=logger, **kwargs)
+        self.locator = locator or ".//article[contains(@class, 'pf-c-card')]"
+
+    PARAMETERS = ("position",)
+
+    ROOT = ParametrizedLocator("{@locator}[{position}]")
+
+
+    def __locator__(self):
+        return self.ROOT
+
+    @classmethod
+    def all(cls, browser):
+        # todo: OUIA versions should return component ids
+        elements = browser.elements('./article')
+        result = []
+        for index, item in enumerate(elements):
+            result.append((index + 1,))
+        return result
+
+
+class CardGroup( GenericLocatorWidget, View):
+    
+    def __init__(self, parent, locator=None, logger=None, **kwargs):
+        View.__init__(self, parent, logger=logger, **kwargs)
+        self.locator = locator or '//section[@class="pf-c-page__main-section"]/div'
```

Comment:
Is there a reason to have the locator be absolute?

Reply:
fixed